frontend/amundsen_application/static/js/pages/SearchPage/SearchFilter/CheckBoxFilter/index.tsx

```diff
@@ -60,26 +59,17 @@ export class CheckBoxFilter extends React.Component<CheckBoxFilterProps> {
 
   onCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     let { checkedValues } = this.props;
-    const { updateFilter } = this.props;
+    const { applyFilters } = this.props;
     const { value } = e.target;
     const categoryId = e.target.name;
 
     if (e.target.checked) {
-      checkedValues = {
-        ...checkedValues,
-        [value]: true,
-      };
+      checkedValues = [...checkedValues, value];
     } else {
-      /* Removing an object key with object destructuring */
-      const { [value]: removed, ...newCheckedValues } = checkedValues;
-      checkedValues = newCheckedValues;
+      checkedValues.splice(checkedValues.indexOf(value), 1);
```

Comment:
We should try to avoid operations like splice that mutate an object.
The previous approach would work better, as it is creating a new object.